src/pages/[slug].astro

```diff
@@ -11,25 +11,14 @@ const { slug } = Astro.params;
 let page;
 
 if (!Astro.isPrerendered) {
```

Comment:
In this if else we are calling the same `fetchSlug` twice. I think we can get drop the if else and just call the fetchSlug and return 404 if it doesn't exist.